Esp8266/Esp8266.ino

```diff
@@ -1,86 +1,116 @@
-#include "event.pb.h"
-
-#include <pb_common.h>
-#include <pb.h>
-#include <pb_encode.h>
-#include <pb_decode.h>
+#include "pb_common.h"
+#include "pb.h"
+#include "pb_encode.h"
+#include "pb_decode.h"
+#include "Esp8266_Interface.pb.h"
 
 #include <ESP8266WiFi.h>
 
-const char* ssid     = "onhub";
-const char* password = "ekchotaghar";
-const char* addr     = "10.0.1.181";
-const uint16_t port  = 10100;
+const char* ssid     = "SSID";
+const char* pass     = "PASS";
+const uint16_t port  = 42069;
 
+WiFiServer server(port);
 WiFiClient client;
 
-void sendStat(pb_Event e);
+struct InputData {
+  static const unsigned buffer_length = esp8266_Esp8266_Command_size;
+  unsigned char buffer[buffer_length];
+  pb_istream_t input_stream;
+  esp8266_Esp8266_Command command;
 
-// setup WIFI and sensor
-void setup() {
-  pinMode(LED_BUILTIN, OUTPUT);
-  Serial.begin(115200);
-  delay(10);
+  InputData() { refresh(); };
 
-  Serial.println();
-  Serial.print("Setting up WIFI for SSID ");
-  Serial.println(ssid);
+  void refresh() {
+    input_stream = pb_istream_from_buffer(buffer, buffer_length);
+    command = esp8266_Esp8266_Command_init_zero;
+  }
+
+  void readAndDecode() {
+    refresh();
+    client.read(buffer, client.available());
+    pb_decode(&input_stream, esp8266_Esp8266_Command_fields, &command);
+  }
+} input;
 
-//  WiFi.mode(WIFI_STA);
-  WiFi.begin(ssid, password);
+struct OutputData {
+  static const unsigned buffer_length = esp8266_Esp8266_Metrics_size;
+  unsigned char buffer[buffer_length];
+  pb_ostream_t output_stream;
+  esp8266_Esp8266_Metrics metrics;
 
-  while (WiFi.status() != WL_CONNECTED) {
-    Serial.println("WIFI connection failed, reconnecting...");
-    delay(500);
+  OutputData() { refresh(); }
```

Comment:
I may be naive, but could I get an explanation to what this does? Why do we need `OutputData()`?

Reply:
It's basically a class I was going to use to simplify the code for dummy_client and Esp8266.ino because they ended up looking very similar. They are pretty much the same, but dummy_client sends command & receives metrics while Esp8266 obviously does the opposite. This code would make it look nicer IMO. You'll see how it is when I'm done. For now, it works.

Reply:
Woody